src/sentry/quotas/redis.py

```diff
@@ -138,15 +143,16 @@ def is_rate_limited(self, project, key=None, timestamp=None):
         if not quotas:
             return NotRateLimited()
 
-        def get_next_period_start(interval):
+        def get_next_period_start(interval, shift):
             """Return the timestamp when the next rate limit period begins for an interval."""
-            return ((timestamp // interval) + 1) * interval
+            return ((timestamp // interval) + 1) * interval - shift
 
         keys = []
         args = []
         for quota in quotas:
-            keys.append(self.__get_redis_key(quota.key, timestamp, quota.window))
-            expiry = get_next_period_start(quota.window) + self.grace
+            shift = project.organization_id % quota.window
```

Comment:
I think an abstraction would be useful if the offset _wasn't_ the highest level identifier we had.